streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -377,9 +409,22 @@ public void setUncaughtExceptionHandler(final Thread.UncaughtExceptionHandler eh
     }
 
     private void validateIsRunning() {
-        if (state != RUNNING) {
+        if (state != KafkaStreamsState.RUNNING) {
             throw new IllegalStateException("KafkaStreams is not running");
         }
     }
 
+    /**
+     * @return The state this instance is in
+     */
+    public KafkaStreamsState getState() {
+        for (final StreamThread thread : threads) {
+            if (thread.state == StreamThread.StreamThreadStateType.PARTITIONS_REVOKED ||
+                thread.state == StreamThread.StreamThreadStateType.PARTITIONS_ASSIGNED) {
+                return KafkaStreamsState.REBALANCING;
```

Comment:
this is a bit racy, we could potentially end up returning RUNNING after we entered a rebalancing state, not sure what kind of guarantees we want to offer here.